Docs: Add links, fix grammar, formatting, wording (#22381)
grafana/grafana: contribute/style-guides/themes.md

## Overview

> **Themes are implemented in Typescript.** That's because our goal is to share variables between Grafana TypeScript and [Sass](https://sass-lang.com/) code. Theme definitions are located in the following files:

Reviewer: ```suggestion
**Themes are implemented in Typescript.** That's because our goal is to share variables between Grafana TypeScript and [SASS](https://sass-lang.com/) code. Theme definitions are located in the following files:
```

Author: "Sass" uses this capitalization. It's a language name like "Java" or "Ruby". Do you mean the the "Sassy CSS" syntax, "SCSS"? 

References:
- https://sass-lang.com/
- https://en.wikipedia.org/wiki/Sass_(stylesheet_language)
- https://www.designernews.co/stories/49077-sass-its-sass-not-sass

- [grafana/packages/grafana-ui/src/themes/dark.ts](../../packages/grafana-ui/src/themes/dark.ts)

> - [grafana/packages/grafana-ui/src/themes/default.ts](../../packages/grafana-ui/src/themes/default.ts)

Reviewer 2: ```suggestion
- [packages/grafana-ui/src/themes/default.ts](../../packages/grafana-ui/src/themes/default.ts)
```

> - [grafana/packages/grafana-ui/src/themes/light.ts](../../packages/grafana-ui/src/themes/light.ts)

Reviewer 2: ```suggestion
- [packages/grafana-ui/src/themes/light.ts](../../packages/grafana-ui/src/themes/light.ts)
```

The `default.ts` file holds common variables like typography and spacing definitions, while `[light|dark].ts` primarily specify colors used in themes.

## Usage

> 

Reviewer: Please avoid stacked headings. Add an intro, even just one sentence, to provide context to what this section is about.

Author: I just added a newline here to separate the headings, but I agree with you. I'm not familiar enough with this documentation to know what should go here. Suggestions?

Reviewer: When I'm not sure, I usually put something like: This section provides usage guidelines.

At least it breaks up the headings, and hopefully a SME will add more later.

Author: Done: https://github.com/grafana/grafana/pull/22381/commits/cfecee995cf2a6aecce1d0b6178ee14204315bb7

> `default.ts` file holds common variables like typography and spacing definitions, while `[light|dark].ts` primarily specify colors used in themes.

Reviewer 2: ```suggestion
The `default.ts` file holds common variables like typography and spacing definitions, while `[light|dark].ts` primarily specify colors used in themes.
```

});
```

### Using themes in [Storybook](https://storybook.js.org/)

> All stories are wrapped with `ThemeContext.Provider` using global decorator. To render `Themeable` component that's not wrapped by `withTheme` HOC you either create a new component in your story:

Reviewer 2: ```suggestion
All stories are wrapped with `ThemeContext.Provider` using a global decorator. To render a `Themeable` component that isn't wrapped by `withTheme` HOC you either create a new component in your story:
```

});
```

### Using themes in [Storybook](https://storybook.js.org/)

> All stories are wrapped with `ThemeContext.Provider` using a global decorator. To render a `Themeable` component that isn't wrapped by `withTheme` HOC, create a new component in your story.

Reviewer 2: I think meaning gets lost with this edit since two alternatives are described below (new component vs `renderComponentWithTheme`):

Author: What do you think, @oddlittlebird ? https://github.com/grafana/grafana/pull/22381#discussion_r383377845

Reviewer 2: Specifically, the "or" on line 95 is orphaned after the edit.

Reviewer: If two components are described, then they need to be mentioned in the line 84 paragraph. The "or" that follows is so far away, it's not clear that it is continuing the previous thought. I would do something like this:

...HOC, you can create a new component in the story or do another thing.

**Thing 1**
Briefly explain Thing 1.
```
do code
```

**Thing 2**
Briefly explain Thing 1.
```
do code
```
---------
You can replace the bold above with headings.

Author: I like this solution better than the original. I'll make this change.

Author: Done: https://github.com/grafana/grafana/pull/22381/commits/75c6b09dc126c7ef240fc4ea23b45eb126e88ba8

> All stories are wrapped with `ThemeContext.Provider` using a global decorator. To render a `Themeable` component that isn't wrapped by `withTheme` HOC you either create a new component in your story:

Reviewer: ```suggestion
All stories are wrapped with `ThemeContext.Provider` using a global decorator. To render a `Themeable` component that isn't wrapped by `withTheme` HOC, create a new component in your story.
```

### Using themes in Angular code

> There should be very few cases where theme would be used in an Angular context. For this purpose, there is a function available that retrieves current theme:

Reviewer 2: ```suggestion
There should be very few cases where a theme would be used in an Angular context. For this purpose, there is a function available that retrieves the current theme:
```

```ts
import { getCurrentTheme } from app/core/utils/ConfigProvider
```

> Angular components should be migrated to React, or if that's not possible at the moment, styled using Sass.

Reviewer: ```suggestion
Angular components should be migrated to React, or if that's not possible at the moment, styled using SASS.
```

## FAQ

### How can I modify Sass variable files?

> For the following to apply you need to run `yarn dev` task.

> `[_variables|_variables.dark|_variables.light].generated.scss` files are the ones that are referenced in the main Sass files for Sass variables to be available. **These files are automatically generated and should never be modified by hand!**

Reviewer: SASS is an acronym. All caps throughout.

Author: I believe it's a backronym... Either way, the language name is currently capitalized like "Sass": https://github.com/grafana/grafana/pull/22381/files/a44bb6cd4f96ed7eb318b8145f7d08004e049289#r383376872

Reviewer 2: The name of the [project](https://sass-lang.com/) is indeed Sass, so I agree with @robatron - I don't see any reason to diverge.

> #### If you need to modify *Sass variable value* you need to modify corresponding Typescript file that is a source of the variables:

Reviewer 2: ```suggestion
#### If you need to modify a *Sass variable value* you need to modify the corresponding Typescript file that is the source of the variables:
```

> #### If you need to modify a *Sass variable value* you need to modify the corresponding Typescript file that is the source of the variables:

Reviewer: ```suggestion
#### If you need to modify a *SASS variable value*, then you need to modify the corresponding TypeScript file that is the source of the variables:
```

Reviewer 2: I agree with the spelling of Sass as discussed above @oddlittlebird.

- `_variables.generated.scss` - modify `grafana-ui/src/themes/default.ts`
- `_variables.light.generated.scss` - modify `grafana-ui/src/themes/light.ts`
- `_variables.dark.generated.scss` - modify `grafana-ui/src/themes/dark.ts`

#### If you need to *add new variable* to Sass variables you need to modify corresponding template file:
- `_variables.generated.scss` - modify `grafana-ui/src/themes/_variables.scss.tmpl.ts`
- `_variables.light.generated.scss` - modify `grafana-ui/src/themes/_variables.light.scss.tmpl.ts`
- `_variables.dark.generated.scss` - modify `grafana-ui/src/themes/_variables.dark.scss.tmpl.ts`

## Limitations

### You must ensure `ThemeContext` provider is available in a React tree

> By default all react2angular directives have `ThemeContext.Provider` ensured. But, there are cases where we create another React tree via `ReactDOM.render`. This happens in case of graph legend rendering and `ReactContainer` directive. In such cases theme consumption will fail. To make sure theme context is available in such cases, you need to wrap your rendered component with ThemeContext.Provider using `provideTheme` function:

Reviewer 2: ```suggestion
By default all react2angular directives have `ThemeContext.Provider` ensured. But, there are cases where we create another React tree via `ReactDOM.render`. This happens in the case of graph legend rendering and the `ReactContainer` directive. In such cases theme consumption will fail. To make sure theme context is available in such cases, you need to wrap your rendered component with ThemeContext.Provider using the `provideTheme` function:
```
